dask_geopandas/core.py

```diff
@@ -429,6 +432,101 @@ def geohash(self, string=True, p=12):
 
         return geohashes
 
+    def spatial_shuffle(
+        self,
+        by="hilbert",
+        p=None,
+        calculate_partitions=True,
+        drop_index=True,
```

Comment:
[drop_index](https://docs.dask.org/en/latest/_modules/dask/dataframe/core.html#DataFrame.set_index) is not necessary because we are not using a column to partition the Dask GeoDataFrame - so there is nothing to drop.

Reply:
This controls whether the original index is dropped or not. If you have some meaningful info there, you may want to keep it. If it's just a range, you can drop it. See how it is passed below to `set_index`.

Reply:
We might still mention that in eg the notes section that this will remove any existing index? 

Do we still want to option to not _set_ the distance as the index? Or you can do a manual `.reset_index(drop=True)` in case you don't want it? 
Eg when writing the result of `spatial_shuffle` to a partitioned (parquet) dataset, you might not necessarily care about the exact distance values for later analyses, and then it just occupies extra space in the files.

Reply:
> We might still mention eg in the notes section that this will remove any existing index?

It is on the line 452.

> Do we still want to option to not set the distance as the index? Or you can do a manual `.reset_index(drop=True)` in case you don't want it?

I would vote for the latter. If we added the keyword, we would in the end only used it to call `.reset_index(drop=True)` under the hood.